src/App.tsx

```diff
@@ -15,36 +15,110 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+const FIELD_ALPHABETIC = 'alphabetically';
+const FIELD_LENGTH = 'length';
+
 export const App: React.FC = () => {
+  const [goods, setGoods] = useState<string[]>(goodsFromServer);
+  const [sortBy, setSortBy] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+
+  const getSortedGoods = (orderBy: string, reverse = false) => {
+    const preparedGoods = [...goodsFromServer];
+
+    switch (orderBy) {
+      case FIELD_ALPHABETIC:
+        preparedGoods.sort((a, b) => a.localeCompare(b));
+        break;
+      case FIELD_LENGTH:
+        preparedGoods.sort((a, b) => a.length - b.length);
+        break;
+      default:
+        break;
+    }
+
+    if (reverse) {
+      preparedGoods.reverse();
+    }
+
+    return preparedGoods;
+  };
+
+  const handleSortAlphabetically = () => {
+    const sorted = getSortedGoods(FIELD_ALPHABETIC, isReversed);
+
+    setGoods(sorted);
+    setSortBy(FIELD_ALPHABETIC);
+  };
+
+  const handleSortByLength = () => {
+    const sorted = getSortedGoods(FIELD_LENGTH, isReversed);
+
+    setGoods(sorted);
+    setSortBy(FIELD_LENGTH);
+  };
+
+  const handleReverse = () => {
+    const reversed = [...goods].reverse();
+
+    setGoods(reversed);
+    setIsReversed(prev => !prev);
+  };
+
+  const handleReset = () => {
+    setGoods(goodsFromServer);
+    setSortBy('');
+    setIsReversed(false);
+  };
+
+  const isInitial =
+    goods.every((item, index) => item === goodsFromServer[index]) &&
+    !isReversed;
```

Comment:
```suggestion
  const isInitial =
    goods.every((item, index) => item === goodsFromServer[index]) &&
    !isReversed;
```
why we need iterate all array, what if we have 10000000 goods?
can we find better way to understand that good is initial?

the absence of filters can tell us that the array is initial?